codec/aminojson/json_marshal_test.go

```diff
@@ -0,0 +1,189 @@
+package aminojson_test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/go-amino"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"pgregory.net/rapid"
+
+	authapi "cosmossdk.io/api/cosmos/auth/v1beta1"
+	"cosmossdk.io/api/cosmos/crypto/ed25519"
+	distapi "cosmossdk.io/api/cosmos/distribution/v1beta1"
+	"github.com/cosmos/cosmos-sdk/codec/aminojson"
+	crypto "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	"github.com/cosmos/cosmos-sdk/testutil/rapidproto"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	disttypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"gotest.tools/v3/assert"
+
+	"github.com/cosmos/cosmos-sdk/codec/aminojson/internal/testpb"
+)
+
+func marshalLegacy(msg proto.Message) ([]byte, error) {
+	cdc := amino.NewCodec()
+	return cdc.MarshalJSON(msg)
+}
+
+func TestAminoJSON_LegacyParity(t *testing.T) {
+	cdc := amino.NewCodec()
+	cdc.RegisterConcrete(authtypes.Params{}, "cosmos-sdk/x/auth/Params", nil)
+	cdc.RegisterConcrete(disttypes.MsgWithdrawDelegatorReward{}, "cosmos-sdk/MsgWithdrawDelegationReward", nil)
+	cdc.RegisterConcrete(&ed25519.PubKey{}, crypto.PubKeyName, nil)
+
+	cases := map[string]struct {
+		gogo   any
```

Comment:
```suggestion
		gogo   gogoproto.Message
```

which is imo slightly better than any